packages/jest-cli/src/snapshot_interactive_mode.js

```diff
@@ -69,12 +81,92 @@ export default class SnapshotInteractiveMode {
     this._pipe.write(messages.filter(Boolean).join('\n') + '\n');
   }
 
+  _drawUIDoneWithSkipped() {
+    this._pipe.write(ansiEscapes.clearScreen);
+    const numPass = this._countPaths - this._testAssertions.length;
+
+    let stats = chalk.bold.dim(
+      pluralize('snapshot', this._countPaths) + ' reviewed',
+    );
+    if (numPass) {
+      stats +=
+        ', ' + chalk.bold.green(pluralize('snapshot', numPass) + ' updated');
+    }
+    if (this._skippedNum) {
+      stats +=
+        ', ' +
+        chalk.bold.yellow(pluralize('snapshot', this._skippedNum) + ' skipped');
+    }
+    const messages = [
+      '\n' + chalk.bold('Interactive Snapshot Result'),
+      ' \u203A ' + stats,
+      '\n' + chalk.bold('Watch Usage'),
+
+      chalk.dim(' \u203A Press ') +
+        'r' +
+        chalk.dim(' to restart Interactive Snapshot Mode.'),
+
+      chalk.dim(' \u203A Press ') +
+        'q' +
+        chalk.dim(' to quit Interactive Snapshot Mode.'),
+    ];
+
+    this._pipe.write(messages.filter(Boolean).join('\n') + '\n');
+  }
+
+  _drawUIDone() {
+    this._pipe.write(ansiEscapes.clearScreen);
+    const numPass = this._countPaths - this._testAssertions.length;
+
+    let stats = chalk.bold.dim(
+      pluralize('snapshot', this._countPaths) + ' reviewed',
+    );
+    if (numPass) {
+      stats +=
+        ', ' + chalk.bold.green(pluralize('snapshot', numPass) + ' updated');
+    }
+    const messages = [
+      '\n' + chalk.bold('Interactive Snapshot Result'),
+      ' \u203A ' + stats,
+      '\n' + chalk.bold('Watch Usage'),
+
+      chalk.dim(' \u203A Press ') +
+        'Enter' +
+        chalk.dim(' to return to watch mode.'),
+    ];
+
+    this._pipe.write(messages.filter(Boolean).join('\n') + '\n');
+  }
+
+  _drawUIOverlay() {
+    this._pipe.write(ansiEscapes.cursorUp(6));
+    this._pipe.write(ansiEscapes.eraseDown);
+
+    if (this._testAssertions.length === 0) {
+      return this._drawUIDone();
+    }
+
+    if (this._testAssertions.length - this._skippedNum === 0) {
+      return this._drawUIDoneWithSkipped();
+    }
+
+    return this._drawUIProgress();
+  }
+
   put(key: string) {
     switch (key) {
       case KEYS.S:
-        const testFilePath = this._testFilePaths.shift();
-        this._testFilePaths.push(testFilePath);
-        this._run(false);
+        if (this._skippedNum === this._testAssertions.length) break;
+
+        this._skippedNum += 1;
+        const testAssertion = this._testAssertions.shift();
+        this._testAssertions.push(testAssertion);
```

Comment:
why a `shift` followed by a `push`?

Reply:
I see it was that way before as well...

Reply:
This is how it was moving the skipped test to the end to circle back around after a run

Reply:
Still want to keep this so that when the mode is restart, the remaining testAssertions array contains all the skipped tests. Added a comment to explain the reason